core/Symbols.h

```diff
@@ -582,6 +582,22 @@ class Symbol final {
     const InlinedVector<Loc, 2> &sealedLocs(const GlobalState &gs) const;
     TypePtr sealedSubclassesToUnion(const GlobalState &ctx) const;
 
+    // Record a required ancestor for this class of module
+    void recordRequiredAncestor(MutableContext ctx, SymbolRef ancestor, Loc loc);
+
+    // Associate a required ancestor with the loc it's required at
+    struct RequiredAncestor {
+        SymbolRef origin; // Only used during transitive lookups
```

Comment:
We reuse the same struct for `requiredAncestors` and `requiredAncestorsTransitive`. In the former, the `origin` is always `this`. In the later, it can be any ancestor of `this`.